internal/resources/fetching/fetchers/gcp/assets_fetcher.go

```diff
@@ -94,33 +79,31 @@ func NewGcpAssetsFetcher(_ context.Context, log *clog.Logger, ch chan fetching.R
 }
 
 func (f *GcpAssetsFetcher) Fetch(ctx context.Context, cycleMetadata cycle.Metadata) error {
-	f.log.Info("Starting GcpAssetsFetcher.Fetch")
-
-	for typeName, assetTypes := range GcpAssetTypes {
-		assets, err := f.provider.ListAllAssetTypesByName(ctx, assetTypes)
-		if err != nil {
-			f.log.Errorf("Failed to list assets for type %s: %s", typeName, err.Error())
-			continue
-		}
-
-		for _, asset := range assets {
-			select {
-			case <-ctx.Done():
-				f.log.Infof("GcpAssetsFetcher.Fetch context err: %s", ctx.Err().Error())
+	f.log.Info("GcpAssetsFetcher.Fetch start")
+	defer f.log.Info("GcpAssetsFetcher.Fetch done")
+
+	resultsCh := make(chan *inventory.ExtendedGcpAsset)
+	go f.provider.ListAssetTypes(ctx, lo.Keys(reversedGcpAssetTypes), resultsCh)
```

Comment:
instead of fetching each asset type separately, we now send a request for all asset types and get back an iterator for all of them